pkg/downloader/manager.go

```diff
@@ -735,7 +745,19 @@ func (m *Manager) findChartURL(name, version, repoURL string, repos map[string]*
 			return
 		}
 	}
-	url, err = repo.FindChartInRepoURL(repoURL, name, version, certFile, keyFile, caFile, m.Getters)
+
+	// Store previously found chart URLs in a map. If repositories share the
+	// same repo URL there is no need to find the same repo URL again. This
+	// improves performance.
+	if baseURL, ok := baseChartUrls[repoURL]; !ok {
+		url, err = repo.FindChartInRepoURL(repoURL, name, version, certFile, keyFile, caFile, m.Getters)
+		if err == nil {
+			slice := strings.SplitAfter(url, "/")
+			baseChartUrls[repoURL] = strings.Join(slice[:len(slice)-1], "")
+		}
+	} else {
+		url = fmt.Sprintf("%s%s-%s.tgz", baseURL, name, version)
```

Comment:
This makes assumptions about the structure of the chart URL, are we sure the assumptions are correct?

An alternative suggestion:
The chart URL is knowable earlier in the flow, in Resolve (resolver.go) at the point where we set the version (inside loop on lines 197-200), the url is available from ver.URLs[0]. We could store (perhaps in a map also returned from the method) the chart URLs which can be used during the download rather than using additional logic to again find the url

Reply:
Cool, yeah makes more sense to do it that way. Have that updated now